server/migrations/20200827191721-reaction-subscriptions.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-disable quotes */
+'use strict';
+
+module.exports = {
+  up: (queryInterface, DataTypes) => {
+    return queryInterface.sequelize.transaction(async (t) => {
+      const users = await queryInterface.sequelize.query(`SELECT * FROM "Users";`, { transaction: t, });
+      await Promise.all(users[0].map(async (user) => {
+        const resSubs = await queryInterface.sequelize.query(`SELECT * FROM "Subscriptions" WHERE subscriber_id=${user.id} AND category_id IN ('new-comment-creation', 'new-reaction');`, { transaction: t, });
+        const subscriptions = resSubs[0];
+
+        const orphanedCommentSubscriptions = [];
+        while (subscriptions.length) {
+          const subscription1 = subscriptions.splice(0, 1)[0];
+          const subscription2 = subscriptions.findIndex((s) => s.object_id === subscription1.object_id);
+          if (subscription2 === -1) {
+            orphanedCommentSubscriptions.push(subscription1);
+          } else {
+            subscriptions.splice(subscription2, 1);
```

Comment:
1) this splice login seems not ideal, is there a better way to loop over subscriptions?

2) how do we know the subscriptions that are returned are in any particular order? as in my other comment, we should be explicit about subtypes of subscriptions

Reply:
resolved in dm?

Reply:
👍